Automattic/wp-calypso
# Modularise editor state

This PR is one of many working on modularising state in Calypso. This one handles editor state.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Apologies for the size of the PR, but I had no way of making it smaller, as all users of editor state need to be updated.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

#### Changes proposed in this Pull Request

* Move editor state from `ui/editor` to a new top-level `editor` entry.
* Modularise editor state.
* Update all code accordingly.

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/log-in` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `editor` key, even if you expand the list of keys. This indicates that the editor state hasn't been loaded yet.
* Open `/read` on the live branch or your local copy of this branch.
* Verify that an `editor` key is present with the editor state. This indicates that the editor state has been loaded.
* Verify that editor-related functionality works normally, particularly in the classic editor. This indicates that I didn't mess up.
